### PR TITLE
[Feature](JsonReader) support sepecify `$.` as root column in json path

### PR DESCRIPTION
**Pull request by eldenmoon**
```
curl --location-trusted -u root:  -T value.json -H "read_json_by_line:true" -H "format:json"  -H "max_filter_ratio:0.5" -H "jsonpaths: [\"$.id\", \"$.entity_id\", \"$.\"]" http://127.0.0.1:8149/api/regression_test/records/_stream_load
```

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by amorynan — APPROVED** [review]
LGTM

File: be/src/vec/exec/format/json/new_json_reader.cpp [hunk 1]
```diff
@@ -1657,7 +1657,19 @@ Status NewJsonReader::_simdjson_write_columns_by_jsonpath(
                 return st;
             }
         }
-        if (i >= _parsed_jsonpaths.size() || st.is<NOT_FOUND>()) {
+        if (i < _parsed_jsonpaths.size() && JsonFunctions::is_root_path(_parsed_jsonpaths[i])) {
+            // Indicate that the jsonpath is "$.", read the full root json object, insert the original doc directly
```
Comment (xiaokang) [inline be/src/vec/exec/format/json/new_json_reader.cpp:1661]: Does this feature, that load all fields in JSON to variant column, match the need for users? I think usually users only need some fields in JSON, not all fields.

Comment (eldenmoon) [reply be/src/vec/exec/format/json/new_json_reader.cpp:1661]: some user need this feature， map all columns to a variant column

**Comment by eldenmoon** [comment]
run buildall

File: regression-test/data/load_p0/stream_load/test_json_load.out [hunk 2]
```diff
@@ -250,3 +250,9 @@ test	k2_value
 
 -- !select29 --
 10	\N
+
+-- !select30 --
+12345	{"k1":12345,"k2":"11111","k3":111111,"k4":[11111]}	{"k1":12345,"k2":"11111","k3":111111,"k4":[11111]}	111111
+12346	{"k1":12346,"k2":"22222","k4":[22222]}	{"k1":12346,"k2":"22222","k4":[22222]}	\N
+12347	{"k1":12347,"k3":"33333","k5":{"k51":1024,"xxxx":[11111]}}	{"k1":12347,"k3":"33333","k5":{"k51":1024,"xxxx":[11111]}}	33333
+12347	{"k1":12347,"k3":"33333","k4":[22222]}	{"k1":12347,"k3":"33333","k4":[22222]}	33333
```
[the remark this answers is not in the document]
Comment (eldenmoon) [reply regression-test/data/load_p0/stream_load/test_json_load.out:234]: done

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**